### PR TITLE
Implement public API

### PR DESCRIPTION
**Pull request by simahawk**
Implements https://github.com/OCA/storage/issues/51

### CODE REVIEW

File: storage_backend/models/storage_backend.py [hunk 1]
```diff
@@ -5,13 +5,54 @@
...
+import inspect
 import logging
+import warnings
 
 from odoo import _, fields, models
 
 _logger = logging.getLogger(__name__)
 
 
+# TODO: useful for the whole OCA?
```
Comment (sebastienbeau) [inline storage_backend/models/storage_backend.py:18]: yes this should be move in a generic tool !

**Comment by sebastienbeau** [comment]
@simahawk Initialy I choose to not have any public api on storage backend in order to avoid exposing this method to the xmlrpc webservice and so to reduce potential security issue (maybe it's useless)
It's will be interesting to have @lmignon opinion

**Review by lmignon — APPROVED** [review]
I like this clean API. Regarding security, indeed it's painful that any non protected methods are available through XML/JSONRPC calls. IMO it's against the python language best practices but we can't change this. In every cases, the security is enforced by ACLs and Record Rules.

**Comment by simahawk** [comment]
> I like this clean API. Regarding security, indeed it's painful that any non protected methods are available through XML/JSONRPC calls. IMO it's against the python language best practices but we can't change this. In every cases, the security is enforced by ACLs and Record Rules.

Same feeling here..

FWIW I've tried to propose this https://github.com/odoo/odoo/issues/62299

**Review by sebastienbeau — APPROVED** [review]
Thanks for your feedback.
It's seem cleaner to later add @api.rpc_disable (if exist)
In any case if the storage backend have the right access right (no read access for unwanted user) the API will be not available.

**Comment by sebastienbeau** [comment]
@simahawk do you have any plan of other refactor in V13 for storage (or search engine) ? I plan to migrate this repo this end of week

File: storage_backend/models/storage_backend.py (hunk 1 above)
Comment (simahawk) [reply storage_backend/models/storage_backend.py:18]: suggestions?
Shall we embrace the proposal by Pedro to have an all in one `odoo-oca-lib`?

**Comment by simahawk** [comment]
> @simahawk do you have any plan of other refactor in V13 for storage (or search engine) ? I plan to migrate this repo this end of week

not that I'm aware of :rofl: :stuck_out_tongue:

File: storage_backend/models/storage_backend.py (hunk 1 above)
Comment (lmignon) [reply storage_backend/models/storage_backend.py:18]: @simahawk Personally I'm against a all in one lib. I prefer a small (even very small) lib with a dedicated scope than one with tons of unrelated stuffs.

**Pull request merged**